### PR TITLE
PD-1372: Changing callout heading to Overline as h3

### PR DESCRIPTION
**Pull request by spark3-mdb**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [PD-1372](https://jira.mongodb.org/browse/PD-1372)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

File: packages/callout/src/Callout.tsx [hunk 1]
```diff
@@ -208,7 +205,20 @@ function Callout({
...
+        <Overline
+          as="h3"
+          className={cx(
+            css`
+              color: inherit;
+              font-size: 12px;
+              font-weight: 500;
+              letter-spacing: 1px;
+              line-height: 16px;
+            `,
```
Comment (spark3-mdb) [inline packages/callout/src/Callout.tsx:217]: Not sure if this text should have one-off styling that isn't based on the theme or other centralized values for font size, weight, etc., but keeping things as is for parity.

File: packages/callout/src/Callout.tsx [hunk 2]
```diff
@@ -208,7 +205,20 @@ function Callout({
           className={headerIconStyle}
           role="presentation"
         />
-        {headerLabels[variant]}
+        <Overline
+          as="h3"
```
Comment (TheSonOfThomp) [inline packages/callout/src/Callout.tsx:209]: Should the overline be `h2`, and the title be `h3`?

File: packages/callout/src/Callout.tsx (hunk 1 above)
Comment (TheSonOfThomp) [reply packages/callout/src/Callout.tsx:217]: The updated `Callout` design files use the same styles as Typography's `Overline` https://www.figma.com/file/4h2VwjCuJUbeZ7hzD2J1rq/LeafyGreen-Design-System?node-id=5821%3A44855

Comment (TheSonOfThomp) [reply packages/callout/src/Callout.tsx:217]: This seems like a change that happened after Shaneeza finished the redesign

**Review by TheSonOfThomp — CHANGES_REQUESTED** [review]
I think the semantic hierarchy should be a bit different. Also the styles in Figma have updated

File: packages/callout/src/Callout.tsx (hunk 1 above)
Comment (spark3-mdb) [reply packages/callout/src/Callout.tsx:217]: It looks like the font-weight and letter-spacing are still different, so I've just kept those 2 lines now.

Comment (TheSonOfThomp) [reply packages/callout/src/Callout.tsx:217]: I think font weight should be `700` still (we need to document this somewhere, but `Semibold` in Figma translates to bold/700 in CSS)

**Review by TheSonOfThomp — APPROVED** [review]
🎸

**Pull request merged**